### PR TITLE
Build Wolfi images

### PR DESCRIPTION
**Pull request by willdollman**
We have packages, we have base images, so let's build some actual runnable images!

This PR is my main branch for building Wolfi-based versions of our first-party docker images.

Progress tracked in https://github.com/sourcegraph/security/issues/447

## Notes for reviewers

Reviewing does not need to be comprehensive (so don't worry about inspecting every single build script or dockerfile!), as I'll be doing a full verification of each image prior to merging. Helpful feedback would be on any patterns you think could be improved.

One area that could use improvement is duplication in the `wolfi-images/` directory - we could use some YAML templating here.

For each image there's a:
- `build-wolfi.sh` alongside the standard `build.sh` script. Changes are minimal, and in most cases just change `docker build -f Dockerfile [...]` to `Dockerfile.wolfi`. In a few cases they remove unneeded build scripts, such as `install-ctags.sh`
- `Dockerfile.wolfi` which contains the bulk of the changes

The apko build definitions are in the `wolfi-images/` directory. Simple images without additional package requirements rely on `wolfi-images/sourcegraph.yaml`. Images that require packaged deps have their own `.yaml` file.

## Test plan

* Green CI
* These changes will not be made live until each image undergoes full validation
- [x] `main-dry-run` prior to merging https://buildkite.com/sourcegraph/sourcegraph/builds/214773#_

### CODE REVIEW

File: cmd/frontend/build-wolfi.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+trap cleanup EXIT
+
+# Environment for building linux binaries
+export GO111MODULE=on
+export GOARCH=amd64
+export GOOS=linux
+export CGO_ENABLED=0
+
+echo "--- go build"
+pkg="github.com/sourcegraph/sourcegraph/cmd/frontend"
```
Comment (jhchabran) [inline cmd/frontend/build-wolfi.sh:20]: You didn't port the bazel build in that one?

**Review by jhchabran — APPROVED** [review]
It's great to finally see this land 😊  

I've noticed that in some cases, the bazel building process is not ported. I think that's fine, because we'll need to update those things anyway because wolfi isn't running musl, so we got to tweak a few of them anyway. 

And we haven't fully landed those either, so we can update them afterward, I don't think it would be wise to delay landing and testing this because of it.

File: cmd/frontend/build-wolfi.sh (hunk 1 above)
Comment (willdollman) [reply cmd/frontend/build-wolfi.sh:20]: I've just been tracking what the build.sh script on `main` does, and for this image it doesn't use bazel - https://github.com/sourcegraph/sourcegraph/blob/main/cmd/frontend/build.sh

Comment (jhchabran) [reply cmd/frontend/build-wolfi.sh:20]: Yeah, that's because in our branch, it's done in a `build-bazel.sh`

**Pull request merged**